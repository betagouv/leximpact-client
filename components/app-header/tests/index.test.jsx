import { createShallow } from "@material-ui/core/test-utils";

import HeaderContainer from "../index";

const CURRENT_NODE_ENV = process.env;

describe("components | app-header | index", () => {
  let shallow;
  beforeAll(() => {
    jest.resetModules();
    process.env = { ...CURRENT_NODE_ENV };
    delete process.env.NODE_ENV;

    const options = { dive: true };
    shallow = createShallow(options);
  });

  afterAll(() => {
    process.env = CURRENT_NODE_ENV;
  });

  describe("snapshot", () => {
    it("doit correspondre quand l'user n'est pas connecté", () => {
      const props = { isUserLogged: false };
      const wrapper = shallow(<HeaderContainer {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });

    it("doit correspondre quand l'user est connecté", () => {
      const props = { isUserLogged: true };
      const wrapper = shallow(<HeaderContainer {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
