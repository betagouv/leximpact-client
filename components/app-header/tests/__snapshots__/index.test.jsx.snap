// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | app-header | index snapshot doit correspondre quand l'user est connecté 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)
    classes={
      Object {
        "root": "HeaderContainer-toolbarRoot-1",
      }
    }
  >
    <WithStyles(Button)
      classes={
        Object {
          "root": "HeaderContainer-menuButtonRoot-2",
        }
      }
      onClick={[Function]}
    >
      <pure(MenuIcon)
        fontSize="small"
      />
       Menu
    </WithStyles(Button)>
    <WithStyles(Typography)
      classes={
        Object {
          "root": "HeaderContainer-titleRoot-3",
        }
      }
      component="div"
    >
      <span>
        <span
          className="HeaderContainer-lighterTitle-4"
        >
          LEXIMPACT 
        </span>
        <span
          className="HeaderContainer-bolderTitle-5"
        >
          POP
        </span>
      </span>
    </WithStyles(Typography)>
    <div />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`components | app-header | index snapshot doit correspondre quand l'user n'est pas connecté 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)
    classes={
      Object {
        "root": "HeaderContainer-toolbarRoot-1",
      }
    }
  >
    <WithStyles(Button)
      classes={
        Object {
          "root": "HeaderContainer-menuButtonRoot-2",
        }
      }
      onClick={[Function]}
    >
      <pure(MenuIcon)
        fontSize="small"
      />
       Menu
    </WithStyles(Button)>
    <WithStyles(Typography)
      classes={
        Object {
          "root": "HeaderContainer-titleRoot-3",
        }
      }
      component="div"
    >
      <span>
        <span
          className="HeaderContainer-bolderTitle-5"
        >
          OPEN 
        </span>
        <span
          className="HeaderContainer-lighterTitle-4"
        >
          LEXIMPACT
        </span>
      </span>
    </WithStyles(Typography)>
    <div />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
